### ISSUE TITLE
[develop] Can't delete selected text with backspace

### ISSUE THREAD

**Issue by sojusnik** [zim-desktop-wiki/zim-desktop-wiki#2528]
**Describe the bug**
After upgrading to `0.75.2-SNAPSHOT-t202311142122~ubuntu23.04.1` I can't delete selected text by using backspace any more, but when nothing is selected and the cursor is placed after a letter, then deleting with backspace is working.

**To Reproduce**
1. select a word or sentence
2. press backspace on your keyboard
3. be astonished

**Expected behavior**
Backspace should delete selected text.

**Environment (please complete the following information):**
 - Version: 0.75.2-SNAPSHOT-t202311142122~ubuntu23.04.1
 - OS: Ubuntu

**Additional context**
When selecting a text and pressing backspace, then the following appears:

```
$ zim -D
DEBUG: Loading config from: <ConfigFile: /home/sojusnik/.config/zim/preferences.conf>
DEBUG: New extendable: <zim.plugins.InsertedObjectTypeMap object at 0x7fae9ce28fd0>
DEBUG: Loading plugin: printtobrowser
DEBUG: Loading plugin: tableeditor
DEBUG: Load extension: <class 'zim.plugins.tableeditor.TableViewObjectType'>
DEBUG: register_object: "table"
DEBUG: Loading plugin: versioncontrol
DEBUG: Loading plugin: bookmarksbar
DEBUG: Loading plugin: spell
DEBUG: Loading plugin: insertsymbol
DEBUG: Loading plugin: pathbar
DEBUG: Loading plugin: backlinkpane
DEBUG: Loading plugin: tags
DEBUG: Loading plugin: linkmap
DEBUG: Loading plugin: attachmentbrowser
DEBUG: Loading plugin: distractionfree
DEBUG: Loading plugin: commandpalette
DEBUG: Loading plugin: tagcomplete
DEBUG: Loading plugin: linesorter
DEBUG: Loading plugin: sourceview
DEBUG: Load extension: <class 'zim.plugins.sourceview.SourceViewObjectType'>
DEBUG: register_object: "code"
DEBUG: Loading plugin: tableofcontents
INFO: This is zim 0.75.2
DEBUG: Python version is sys.version_info(major=3, minor=11, micro=4, releaselevel='final', serial=0)
DEBUG: Platform is posix
DEBUG: Not running from a source dir
DEBUG: Set XDG_DATA_HOME to /home/sojusnik/.local/share
DEBUG: Set XDG_DATA_DIRS to [<LocalFolder: /usr/share/gnome-xorg>, <LocalFolder: /usr/share/gnome>, <LocalFolder: /home/sojusnik/.local/share/flatpak/exports/share>, <LocalFolder: /var/lib/flatpak/exports/share>, <LocalFolder: /usr/local/share>, <LocalFolder: /usr/share>, <LocalFolder: /var/lib/snapd/desktop>]
DEBUG: Set XDG_CONFIG_HOME to /home/sojusnik/.config
DEBUG: Set XDG_CONFIG_DIRS to [<LocalFolder: /etc/xdg/xdg-gnome-xorg>, <LocalFolder: /etc/xdg>]
DEBUG: Set XDG_CACHE_HOME to /home/sojusnik/.cache
DEBUG: Connecting to /run/user/1000/zim-0.75.2-e63203e4
DEBUG: Got error in dispatch: No such file or directory
DEBUG: Starting primary process
DEBUG: Start listening on: /run/user/1000/zim-0.75.2-e63203e4
DEBUG: Loading config from: <zim.notebook.info.VirtualFile object at 0x7fae990e51d0>
DEBUG: Loading config from: /home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines/notebook.zim
DEBUG: Loading config from: /home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines/notebook.zim
DEBUG: Connecting to database file: /home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines/.zim/index.db
DEBUG: Loading config from: /home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines/.zim/state.conf
DEBUG: New extendable: <Notebook: Allgemeines>
DEBUG: Load extension: <class 'zim.plugins.versioncontrol.VersionControlNotebookExtension'>
INFO: VCS detected: git - /home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines
DEBUG: Loading config from: <ConfigFile: /home/sojusnik/.config/zim/style.conf>
DEBUG: Autosave interval: 10 - use threads: True
INFO: Page changed on disk: Produktivität:Zeitmanagement:Aufgaben
INFO: Open page: Produktivität:Zeitmanagement:Aufgaben (Produktivität:Zeitmanagement:Aufgaben)
DEBUG: New extendable: <notebookview.NotebookView object at 0x7fae98ae47c0 (zim+gui+notebookview+NotebookView at 0x1b8d710)>
DEBUG: Load extension: <class 'zim.plugins.attachmentbrowser.AttachmentBrowserWindowExtension'>
DEBUG: Action: toggle_panes(False)
DEBUG: Action: toggle_panes(True)
DEBUG: Load extension: <class 'zim.plugins.backlinkpane.BackLinksPanePageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.insertsymbol.InsertSymbolPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.linesorter.LineSorterPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.linkmap.LinkMapPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.printtobrowser.PrintToBrowserPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.spell.SpellPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.tableofcontents.ToCPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.tags.TagsNotebookViewExtension'>
DEBUG: New extendable: <mainwindow.MainWindow object at 0x7fae9e1cb400 (zim+gui+mainwindow+MainWindow at 0x1b3ffd0)>
DEBUG: Load extension: <class 'zim.plugins.bookmarksbar.BookmarksBarMainWindowExtension'>
DEBUG: Load extension: <class 'zim.plugins.commandpalette.CommandPaletteMainWindowExtension'>
DEBUG: Load extension: <class 'zim.plugins.distractionfree.DistractionFreeMainWindowExtension'>
DEBUG: Load extension: <class 'zim.plugins.pathbar.PathBarMainWindowExtension'>
DEBUG: Load extension: <class 'zim.plugins.tagcomplete.TagAutocompleteWin'>
DEBUG: Load extension: <class 'zim.plugins.versioncontrol.VersionControlMainWindowExtension'>
INFO: Running: ('git', 'status', '--porcelain') (cwd: '/home/sojusnik/Sonstiges/Zim Notebooks/Allgemeines')
DEBUG: Accelmap: /home/sojusnik/.config/zim/accelmap
DEBUG: No autosave version needed - no changes
DEBUG: Add window: MainWindow
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/zim/gui/pageview/textview.py", line 377, in do_key_press_event
    run_post, handled = self._do_key_press_event(keyval, event_state)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/zim/gui/pageview/textview.py", line 415, in _do_key_press_event
    return False, self._do_key_press_event_selection(keyval, event_state)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/zim/gui/pageview/textview.py", line 623, in _do_key_press_event_selection
    handled = decrement_indent(start, end)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/zim/gui/pageview/textview.py", line 574, in decrement_indent
    if buffer.get_range_in_verbatim_block(start, end):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/zim/gui/pageview/textbuffer.py", line 1787, in get_range_in_verbatim_block
    t = filter(_is_pre_tag, iter.get_tags())
                            ^^^^^^^^^^^^^
AttributeError: 'builtin_function_or_method' object has no attribute 'get_tags'
```

**Comment by bruvellu** [comment]
This is also happening to me on 0.75.2-SNAPSHOT-t202401091956~ubuntu22.04.1. It works if I use DELETE, though.

[Screencast from 16-01-24 12:23:48.webm](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/361179/28babdd5-b4fe-4f37-b15b-3c95705259e4)

**Comment by introt** [comment]
Fixed by 41d1125f567b984b36a215826abba90735eaa068
